### DISCUSSION TITLE
✔️Load and Use with AFK Arena Test Server

### DISCUSSION THREAD

**Discussion by AraloreVlex** [zebscripts/AFK-Daily#66] [Ideas]
I would like to see this be able to be used with AFK Arena Test Server. I know this might not be as simple as I might think because I have little to no knowledge of the scripting language. Maybe an option in the config to choose main or test server.

**Comment by AraloreVlex** [comment]
Here's my idea from some testing I did.

In config.sh add:
     ` testServer=false # Set to true if you wish to load test server.`

Now in afk-daily.sh

For Start App
      

> function startApp() {
>     if [ "$testServer" == true ]; then
>             monkey -p com.lilithgames.hgame.gp.id 1 >/dev/null 2>/dev/null
>             sleep 1
>             disableOrientation
>     else
>            monkey -p com.lilithgame.hgame.gp 1 >/dev/null 2>/dev/null
>            sleep 1
>            disableOrientation
>     fi
> }

For Closes the app

> function closeApp() {
> 	if [ "$testServer" == true ]; then
> 		am force-stop com.lilithgames.hgame.gp.id >/dev/null 2>/dev/null
> 	else
> 		am force-stop com.lilithgame.hgame.gp >/dev/null 2>/dev/null
>     fi
> }


On test with false it closed main server app if open and opened it if closed and with true it did the same but with the test server app.

Further testing will be needed to see if all functions work ass intended with the test server app.

**Reply by Zebiano** [reply]
Nice idea and nice testing. Though I do not own a test server account, and even if I would own one, I'd have to spend some time leveling it up... Not to mention I'd have to start testing the script on two different server types, which is really something I do not want, nor do I think the script *needs* this feature.

But, I do not mind adding this feature, as I do believe the script should still work on the test server, unless something big gets changed. I'd just have to put a big warning saying **use at your own risk**, as none of the features will ever be tested on the test server.

May I ask why you'd like to have this on the test server? Since it's a test server, there's not really any point in trying to progress in it that much...?

**Reply by AraloreVlex** [reply]
I play one account on the Main server and 1 on the test server. Mostly this was just me learning how the code works and what the different things do. I was wanting to automate the test account to catch it up to my main account while I put time into the main account. Plus theres that At own risk thing> If was to be banned for use of a script or what even I would prefer it be my F2P test account and not the P2W main lol.

So I played with code to see what was doing what and since it got it to work that far I figured I would post the idea and what I did. I will get into further testing own my own in a day or too and if works as should will inform. 

Then it will just be and option for people to have just incase :)

**Reply by kevingrillet** [reply]
I think we could add the functionality as a "secret" one, not displayed in the default `config` file because it's experimental (not tested, no liability, no warranty), but maybe appear in the readme/wiki?

**Reply by Zebiano** [reply]
> I think we could add the functionality as a "secret" one, not displayed in the default `config` file because it's experimental (not tested, no liability, no warranty), but maybe appear in the readme/wiki?

I was thinking of making a flag (maybe `-t`) for this instead of a `config.sh` variable. 

I'm not sure yet if I want to keep it a "secret". That also crossed my mind when I read this idea... Though it would not make sense to implement things that are meant for the end user, but then not let the end user know they're implemented... We could maybe not write it right at the beginning of the `README.md`, but rather at a more discrete place inside it. I think that's the way to go in this case.

**Reply by kevingrillet** [reply]
Need to be tested, but [done](https://github.com/zebscripts/AFK-Daily/pull/68).

**Comment by delgatojr** [comment]
There's a test server? Is it like unlimited currency?

**Reply by AraloreVlex** [reply]
Yes there is just google afk arena Test Server.  It works same as main server except Champaign difficulty is lowered a little and new evens, heroes, and other things are added about a week before they are added to the main server for testing and such. Other then that its no different except cost of stuff in the market is a little more expensive. 

All in all this just a thought :)

**Comment by AraloreVlex** [comment]
The script runs fine up to the Arena of Heroes bit then it stalls. This may be due to the fact that I dont have the skip button unlocked yet. But will see.  Will work on getting it unlocked and test again

Turning arena of heroes feature off in config and the script continues flawlessly. Once I get skip unlocked I can test that part.

**Reply by Zebiano** [reply]
What do you mean by "stalls"? If it just waits there and doesn't do anything, try finishing the fight and see if the script picks up from there. If not, then there might be something wrong with the script indeed, or the fact that it's a test server and the game works slightly differently there.

**Reply by AraloreVlex** [reply]
Yea it just waits and does nothing. It will require further testing once I have able to unlock the skip button in arena so it can be proper tested. Other then that everything runs fine.
